vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/ActionSteps.java

```diff
@@ -16,90 +16,74 @@
 import org.vividus.bdd.monitor.TakeScreenshotOnFailure;
-import org.vividus.bdd.steps.ui.web.model.Action;
 import org.vividus.bdd.steps.ui.web.model.ActionType;
 import org.vividus.bdd.steps.ui.web.validation.IBaseValidations;
 import org.vividus.selenium.IWebDriverProvider;
-import org.vividus.softassert.ISoftAssert;
 import org.vividus.ui.web.action.search.SearchAttributes;
 
 @TakeScreenshotOnFailure
 public class ActionSteps
 {
-    @Inject private IWebDriverProvider webDriverProvider;
-    @Inject private IBaseValidations baseValidations;
-    @Inject private ISoftAssert softAssert;
+    private static final String ARGUMENT = "argument";
+
+    private final IWebDriverProvider webDriverProvider;
+    private final IBaseValidations baseValidations;
+
+    public ActionSteps(IWebDriverProvider webDriverProvider, IBaseValidations baseValidations)
+    {
+        this.webDriverProvider = webDriverProvider;
+        this.baseValidations = baseValidations;
+    }
 
     /**
      * Executes the sequence of web actions
      * <div>Example:</div>
      * <code>
      *   <br>When I execute the sequence of actions:
-     *   <br>|type           |searchAttributes                        |offset   |
-     *   <br>|MOVE_BY_OFFSET |                                        |(-300, 0)|
-     *   <br>|MOVE_BY_OFFSET |                                        |(0, 40)  |
-     *   <br>|CLICK_AND_HOLD |By.xpath(//signature-pad-control/canvas)|         |
-     *   <br>|MOVE_BY_OFFSET |                                        |(0, 100) |
-     *   <br>|RELEASE        |By.xpath(//signature-pad-control/canvas)|         |
+     *   <br>|type          |argument                                |
+     *   <br>|DOUBLE_CLICK  |By.xpath(//signature-pad-control/canvas)|
+     *   <br>|CLICK_AND_HOLD|By.xpath(//signature-pad-control/canvas)|
+     *   <br>|MOVE_BY_OFFSET|(-300, 0)                               |
+     *   <br>|RELEASE       |By.xpath(//signature-pad-control/canvas)|
+     *   <br>|SEND_KEYS     |Text                                    |
+     *   <br>|CLICK         |By.xpath(//signature-pad-control/canvas)|
      * </code>
      * <br>
      * <br>
      * where
      * <ul>
-     * <li><code>type</code> is one of web actions: move by offset, click and hold, release</li>
-     * <li><code>searchAttributes</code> is search attribute to find element for interaction
-     *  (could be empty if not applicable)</li>
-     * <li><code>offset</code> the offset to move by (ex.: (10, 10))</li>
+     * <li><code>type</code> is one of web actions: DOUBLE_CLICK, CLICK_AND_HOLD, MOVE_BY_OFFSET,
+     * RELEASE, SEND_KEYS, CLICK</li>
+     * <li><code>argument</code> either text or search attribute or point</li>
      * </ul>
      * @param actions table of actions to execute
      */
     @When("I execute the sequence of actions: $actions")
-    public void executeActionsSequence(List<Action> actions)
+    public void executeActionsSequence(ExamplesTable actions)
```

Comment:
we can't change step API: deprecate the old step, introduce a new step with updated syntax